flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java

```diff
@@ -591,7 +594,13 @@ public void updateHostNameIPMapping(@RequestBody AssetIPMapping mapping) {
          pageSize = FlowgateConstant.maxPageSize;
       }
       PageRequest pageRequest = new PageRequest(pageNumber - 1, pageSize);
-      return assetIPMappingRepository.findAll(pageRequest);
+      if(ip == null) {
+         return assetIPMappingRepository.findAll(pageRequest);
+      }else if(IPAddressUtil.isValidIp(ip)) {
+         return assetIPMappingRepository.findByIp(ip, pageRequest);
+      }
+      throw new WormholeRequestException(HttpStatus.INTERNAL_SERVER_ERROR, "Invalid ip address",
```

Comment:
add the ip address in the return message.

Reply:
Done